src/server-extension/resolvers/ChannelsResolver/utils.ts

```diff
@@ -105,3 +105,76 @@ export function buildExtendedChannelsQuery(
 
   return listQuery
 }
+
+export function buildTopSellingChannelsQuery(
+  args: TopSellingChannelsArgs,
+  info: GraphQLResolveInfo,
+  ctx: Context
+) {
+  const roundedDate = new Date().setMinutes(0, 0, 0)
+
+  const tree = getResolveTree(info)
+
+  // Extract subsquid-supported Channel sql args
+  const sqlArgs = parseSqlArguments(model, 'Channel', {
+    ...args,
+    where: args.where?.channel, // only supported WHERE part
+  })
+
+  // Extract subsquid-supported Channel fields
+  const channelSubTree = tree.fieldsByTypeName.TopSellingChannelsResult.channel
+  const channelFields = parseAnyTree(model, 'Channel', info.schema, channelSubTree)
+
+  // Generate query using subsquid's ListQuery
+  const listQuery = new ListQuery(model, ctx.openreader.dialect, 'Channel', channelFields, sqlArgs)
+  let listQuerySql = listQuery.sql
+
+  // Count NFT sells from the auctions and buy now events and add them to the query
+  listQuerySql = extendClause(listQuerySql, 'SELECT', '"top_selling_channels"."amount"')
+
+  listQuerySql = extendClause(
+    listQuerySql,
+    'FROM',
+    `
+    INNER JOIN (
+      SELECT
+        (SUM((COALESCE(event.data->>'price', '0')::bigint)) + SUM(COALESCE(winning_bid.amount, 0))) AS "amount",
+        "data"->'previousNftOwner'->>'channel' AS "channel_id"
+      FROM "event"
+      LEFT JOIN bid AS winning_bid ON "data"->>'winningBid' = winning_bid.id
+      WHERE
+      ${
+        args?.periodDays > 0 &&
+        ` "event"."timestamp" > '${new Date(
+          roundedDate - args.periodDays * 24 * 60 * 60 * 1000
+        ).toISOString()}' AND`
+      }
```

Comment:
There's a syntax error if you provide `periodDays == 0`, because `args?.periodDays > 0 && ...` will then evaluate to `false` and be converted to `'false'` string. You should use short-if syntax instead:
```suggestion
      ${
        args?.periodDays > 0
          ? ` "event"."timestamp" > '${new Date(
              roundedDate - args.periodDays * 24 * 60 * 60 * 1000
            ).toISOString()}' AND`
          : ''
      }
```